### PR TITLE
[CDAP-3507] Feature/ui validator transform

### PR DESCRIPTION
**Pull request by elfenheart**
TODO:
- Save JSON blob to backend to retrieve the original rules [DONE]
- Add error dataset diagram to the DAG

validator

### CODE REVIEW

File: cdap-ui/app/directives/validators/validators.less [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      border: 2px solid #999999; border-radius: 4px;
+      transition: all 0.3s ease-in 0s;
+  }
+  .onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-inner {
+      margin-left: 0;
+  }
+  .onoffswitch-checkbox:checked + .onoffswitch-label .onoffswitch-switch {
+      right: 0px;
+  }
+
```
Comment (anwar6953) [inline cdap-ui/app/directives/validators/validators.less]: extra newlines here, and missing a newline at end of file.

File: cdap-ui/app/services/app/my-app-dag-service.js [hunk 2]
```diff
@@ -677,6 +682,8 @@ angular.module(PKG.name + '.services')
         if (myHelpers.objectQuery(transform, 'properties') &&
             Object.keys(transform.properties).length > 0) {
           transform.properties = propertiesIterator(transform.properties, transform._backendProperties);
+
```
Comment (anwar6953) [inline cdap-ui/app/services/app/my-app-dag-service.js]: why these extra newlines

**Comment by elfenheart** [comment]
build - http://builds.cask.co/browse/CDAP-DRC2631-1/

**Comment by ajainarayanan** [comment]
:+1: LGTM

**Pull request merged**